src/runtime/wasm.ts

```diff
@@ -642,6 +678,32 @@ export class WasmParticle extends Particle {
     void collection.clear();
   }
 
+  // Called by particles to retrieve the entity held by a reference-typed handle.
+  async dereference(wasmHandle: WasmAddress, refIdPtr: WasmAddress, continuationId: number) {
+    const handle = this.getHandle(wasmHandle);
+    const converter = this.converters.get(handle);
+    if (!converter) {
+      throw new Error('cannot find handle ' + handle.name);
+    }
+    const refId = this.container.read(refIdPtr);
+
+    let ref;
+    if (handle instanceof Singleton) {
+      ref = await handle.get() as Reference;
+    } else if (handle instanceof Collection) {
+      ref = await handle.get(refId);
+    } else {
+      throw new Error(`wasm particle '${this.spec.name}' dereferenced an unsupported handle type ${handle._id}`);
+    }
+    let p = 0;
```

Comment:
Is there a better name you could use than p? What is it? Particle?

Reply:
It's for pointer allocated in the wasm memory, and I've used simple short names for these all through this code. They're only short-lived, typically freed a couple of lines after allocation.

Reply:
ptr?